src/base/Cellar.sol

```diff
@@ -598,6 +662,44 @@ contract Cellar is ERC4626, Ownable, Multicall {
         _depositTo(holdingPosition, assets);
     }
 
+    function deposit(uint256 assets, address receiver) public override returns (uint256 shares) {
+        uint256 _totalAssets = totalAssets();
+
+        _takePerformanceFees(_totalAssets);
+        // Check for rounding error since we round down in previewDeposit.
+        require((shares = _convertToShares(assets, _totalAssets)) != 0, "ZERO_SHARES");
```

Comment:
I know this is just from ERC4626, but could you use refactor this into a custom error using `USR_ZeroShares`?